pkgs/json_syntax_generator/lib/src/generator/helper_library.dart

```diff
@@ -203,6 +203,19 @@ class JsonReader {
     return "Unexpected value '$value' (${value.runtimeType}) for '$pathString'."
         ' Expected a $expectedType.';
   }
+
+  /// Traverses a JSON path, returns `null` if the path cannot be traversed.
+  Object? tryTraverse(List<String> path) {
+    Object? json = this.json;
+    while (path.isNotEmpty) {
+      final key = path.removeAt(0);
```

Comment:
It's slightly inefficient to remove the first element from the list (`O(N^2)`). Why not just loop over the path?